<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Automated Viral Plaque Assay &mdash; Abhineet Ram 0.0.1 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="_static/doctools.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="MCF10A RNA sequencing" href="RNAseq.html" />
    <link rel="prev" title="Spectral Unmixing" href="unmixing.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            Abhineet Ram
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">About me:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="aboutme.html">Abhineet</a></li>
<li class="toctree-l1"><a class="reference internal" href="beforeyoubegin.html">Resume</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Data and Image Analysis:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="ERKETG.html">Deciphering the History of ERK Activity from Fixed-Cell Immunofluorescence Measurements</a></li>
<li class="toctree-l1"><a class="reference internal" href="unmixing.html">Spectral Unmixing</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Automated Viral Plaque Assay</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#workflow">Workflow</a></li>
<li class="toctree-l2"><a class="reference internal" href="#nuclear-segmentation">Nuclear segmentation</a></li>
<li class="toctree-l2"><a class="reference internal" href="#acknowledgements">Acknowledgements</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Bioinformatics:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="RNAseq.html">MCF10A RNA sequencing</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Tutorials:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="PCA.html">Principal Component Analysis</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Abhineet Ram</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Automated Viral Plaque Assay</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/Plaque_assay.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="automated-viral-plaque-assay">
<span id="plaqueassay"></span><h1>Automated Viral Plaque Assay<a class="headerlink" href="#automated-viral-plaque-assay" title="Permalink to this heading"></a></h1>
<figure class="align-default" id="id1">
<img alt="_images/well5_merge.PNG" src="_images/well5_merge.PNG" />
<figcaption>
<p><span class="caption-text">Kidney cells (blue) infected with influenza, resulting in viral plaques (green). 9 x 9 stiched image of 20X images captured on a Nikon Ti-2 eclipse fluorescence microscope.</span><a class="headerlink" href="#id1" title="Permalink to this image"></a></p>
</figcaption>
</figure>
<p>Flu researchers in the Diaz-Munoz Lab at UC Davis often conduct “Plaque Assays,” where cells in culture are exposed to various Flu viruses.
The virus infects cells and then spreads to nearby cells and this results in “plaques” being formed.</p>
<p>Traditionally, the quantification of these plaques relied on manual counting, a labor-intensive process susceptible to human error.
To address this limitation, I developed a MATLAB pipeline that revolutionizes the plaque assay methodology.</p>
<p>This script not only automates plaque counting but also goes beyond, extracting comprehensive properties for each plaque.
It analyzes crucial aspects such as area, the number of infected cells, and the intensity of viral protein expression.
This MATLAB pipeline significantly enhances the precision and efficiency of the assay, bringing it into the 21st century.
This method not only simplifies the research process but also allows for a better understanding of viral infection dynamics.</p>
<section id="workflow">
<h2>Workflow<a class="headerlink" href="#workflow" title="Permalink to this heading"></a></h2>
<figure class="align-default" id="id2">
<img alt="_images/Sidebyside_merge.png" src="_images/Sidebyside_merge.png" />
<figcaption>
<p><span class="caption-text">We will be looking at the whole image and a zoomed in region of the image.</span><a class="headerlink" href="#id2" title="Permalink to this image"></a></p>
</figcaption>
</figure>
<p>Immunofluorescence images of Nucleoprotein (green) and Hoescht (blue) staining are processed
to create object masks. The number of nuclei within each mask are then
counted using nuclear segmentation of the Hoescht image. To create detectable objects,
images are binarized (imbinarize) with a user-defined threshold, followed by the removal of
small, noisy pixels (bwareaopen). The masks are then dilated and filled (imdilate and imfill)
to smoothen them and ensure they do not contain holes. Next, masks are eroded to undo
the dilation step (imerode).</p>
<figure class="align-default">
<img alt="_images/workflow.png" src="_images/workflow.png" />
</figure>
<p>To count the number of cells in viral cluster, object masks
were overlaid with the nuclear segmentation resulting from Hoechst-based segmentation.
Incorrect masks were filtered out by thresholding the min/max mask area and removing
masks that did not contain any nuclei. Finally, each mask are counted and characterized to
obtain pixel intensities (viral protein expression) and object areas.</p>
<figure class="align-default" id="id3">
<img alt="_images/Final_segmented_islands.png" src="_images/Final_segmented_islands.png" />
<figcaption>
<p><span class="caption-text">YFP channel image: counts overlayed onto each plaque</span><a class="headerlink" href="#id3" title="Permalink to this image"></a></p>
</figcaption>
</figure>
<p>If we zoom into the image, we can visualize the mask created by the object segmentation.</p>
<figure class="align-default" id="id4">
<img alt="_images/Final_segmented_islands_zoom.png" src="_images/Final_segmented_islands_zoom.png" />
<figcaption>
<p><span class="caption-text">YFP channel image: Plaque (green) and mask overlay (red)</span><a class="headerlink" href="#id4" title="Permalink to this image"></a></p>
</figcaption>
</figure>
</section>
<section id="nuclear-segmentation">
<h2>Nuclear segmentation<a class="headerlink" href="#nuclear-segmentation" title="Permalink to this heading"></a></h2>
<p>To count the number of cells in each plaque, nuclear segmentaion is conducted in the blue channel image (Hoescht stain).</p>
<table class="docutils align-default">
<tbody>
<tr class="row-odd"><td><figure class="align-default" id="id5">
<img alt="_images/Nuclear_masks.png" src="_images/Nuclear_masks.png" />
<figcaption>
<p><span class="caption-text">All segmented nuclear maks (red)</span><a class="headerlink" href="#id5" title="Permalink to this image"></a></p>
</figcaption>
</figure>
</td>
<td><figure class="align-default" id="id6">
<img alt="_images/Nuclear_masks_zoom.png" src="_images/Nuclear_masks_zoom.png" />
<figcaption>
<p><span class="caption-text">Zoomed in to our region of interest</span><a class="headerlink" href="#id6" title="Permalink to this image"></a></p>
</figcaption>
</figure>
</td>
</tr>
</tbody>
</table>
<p>Then the plaque masks are overlayed onto the segmented nuclear masks, and the number of cells in mask is counted.</p>
<figure class="align-default" id="id7">
<img alt="_images/Final_segmented_islandmasks_with_nucoverlay_zoom.png" src="_images/Final_segmented_islandmasks_with_nucoverlay_zoom.png" />
<figcaption>
<p><span class="caption-text">Nuclear masks (red) ovelayed onto plaque masks (green). Number of cells within each mask are indicated.</span><a class="headerlink" href="#id7" title="Permalink to this image"></a></p>
</figcaption>
</figure>
<p>This script allows us to count and properly quantify the viral palques in each image!
It will output the number of clusters, location area and intensity of each cluster, and the number of cells in each cluster in a table format.</p>
</section>
<section id="acknowledgements">
<h2>Acknowledgements<a class="headerlink" href="#acknowledgements" title="Permalink to this heading"></a></h2>
<p>This experimental and computational pipeline was developed in collaboration with the Diaz-Munoz lab.
Specifically, Illechukwu Agu devloped the experimental Plaque assay with viral protein staining, and also captured the images.</p>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="unmixing.html" class="btn btn-neutral float-left" title="Spectral Unmixing" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="RNAseq.html" class="btn btn-neutral float-right" title="MCF10A RNA sequencing" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, Abhineet Ram.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>